pkg/utils/context.go

```diff
@@ -45,27 +51,35 @@ var (
 	}
 )
 
-// SendingContext creates the context to use when sending a Cloud Event with ceclient.Client. It
-// sets the target and attaches a filtered set of headers from the initial request.
-func SendingContext(ctx context.Context, tctx cloudevents.HTTPTransportContext, targetURI *url.URL) context.Context {
-	sendingCTX := cloudevents.ContextWithTarget(ctx, targetURI.String())
+// ContextFrom creates the context to use when sending a Cloud Event with cloudevents.Client. It
+// sets the target if specified, and attaches a filtered set of headers from the initial request.
+func ContextFrom(tctx cloudevents.HTTPTransportContext, targetURI *url.URL) context.Context {
+	// Get the allowed set of headers.
+	h := PassThroughHeaders(tctx.Header)
+	// Override the headers.
+	tctx.Header = h
+	// Create the sending context with the overridden transport context.
+	sendingCTX := cehttp.WithTransportContext(context.Background(), tctx)
```

Comment:
Should this accept the context as an argument, rather than setting it to Background here?

Reply:
This is a problem when I introduced the multiple value headers... If there is a response that changes a header value, it will get added, thus more than 1 value for a header will be possible... Let's sync more in person also

Reply:
Actually, it might be an SDK bug that I introduced sometime back... the responseContext is created from the send context... And it not a new one, thus headers get added and that can cause a problem

Reply:
Agreed we will do a follow up PR